spec/validators/redirect_uri_validator_spec.rb

```diff
@@ -56,19 +55,45 @@
     expect(subject.errors[:redirect_uri].first).to eq("cannot contain a fragment.")
   end
 
+  it "is invalid when scheme resolves to localhost (needs an explict scheme)" do
+    subject.redirect_uri = "localhost:80"
+    expect(subject).to be_invalid
+    expect(subject.errors[:redirect_uri].first).to eq("a scheme must be specified.")
```

Comment:
Let's use I18n key to test an actual error in order not to change the text everytime :)